pkg/apis/core/validation/shoot.go

```diff
@@ -2582,31 +2580,35 @@ func validateShootOperationContext(operation string, shoot *core.Shoot, fldPath
 }
 
 // ValidateForceDeletion validates the addition of force-deletion annotation on the Shoot.
-func ValidateForceDeletion(newShoot, oldShoot *core.Shoot) *field.Error {
+func ValidateForceDeletion(newShoot, oldShoot *core.Shoot) field.ErrorList {
 	var (
 		fldPath               = field.NewPath("metadata", "annotations").Key(v1beta1constants.AnnotationConfirmationForceDeletion)
+		allErrs               = field.ErrorList{}
 		oldNeedsForceDeletion = helper.ShootNeedsForceDeletion(oldShoot)
 		newNeedsForceDeletion = helper.ShootNeedsForceDeletion(newShoot)
 	)
 
 	if !newNeedsForceDeletion && oldNeedsForceDeletion {
-		return field.Forbidden(fldPath, "force-deletion annotation cannot be removed once set")
+		allErrs = append(allErrs, field.Forbidden(fldPath, "force-deletion annotation cannot be removed once set"))
+		return allErrs
 	}
 
 	if newNeedsForceDeletion && !oldNeedsForceDeletion {
 		if newShoot.DeletionTimestamp == nil {
-			return field.Forbidden(fldPath, "force-deletion annotation cannot be set when Shoot deletionTimestamp is nil")
+			allErrs = append(allErrs, field.Forbidden(fldPath, "force-deletion annotation cannot be set when Shoot deletionTimestamp is nil"))
```

Comment:
Maybe we should return here and not club that error code validation since the user can't control that and if the deletion timestamp is not there IMO we should not give an error code error. As getting that error one can get a sense that they need to change something in the status.

Reply:
I still feel it makes to return both the errors at once instead of first erroring about deletiontimestamp and the user fixes it and the next error is reported.

Reply:
/approve